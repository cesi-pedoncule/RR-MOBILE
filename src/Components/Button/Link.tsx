import React from "react";
import { Text, TouchableOpacity } from "react-native";

import CommonStyles from "../../Styles/CommonStyles";
import LinkStyles from "../../Styles/Components/Button/LinkStyles";

interface Props {
    label: string;
    callBack: () => void;
}

export default function Link({ label, callBack }: Props)
{
    return (
        <TouchableOpacity onPress={callBack} style={LinkStyles.btnBackground}>
            <Text style={LinkStyles.link}>{label}</Text>
        </TouchableOpacity>
    )
}