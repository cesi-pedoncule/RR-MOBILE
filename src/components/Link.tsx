import React from "react";
import { Text, TouchableHighlight } from "react-native";
import commonStyles from "../styles/commonStyles";

interface Props {
    label: string;
    callBack: () => void;
}

export default function Link({label, callBack}: Props)
{
    return (
        <TouchableHighlight onPress={callBack} underlayColor={"#FFFFFF"}>
            <Text style={commonStyles.link}>{label}</Text>
        </TouchableHighlight>
    )
}