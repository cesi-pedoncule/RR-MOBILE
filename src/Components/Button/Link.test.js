import { React } from "react";
import renderer from "react-test-renderer";
import Link from "./Link";

describe("<Link />", () => {
    it("has 1 child", () => {
        const tree = renderer.create(<Link 
            label="testValue" 
            callBack={() => console.log('test')} 
        />).toJSON();
        expect(tree.children.length).toBe(1);
    });

    it("renders correctly", () => {
        const tree = renderer.create(<Link label="testValue" callBack={() => console.log('test')} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
