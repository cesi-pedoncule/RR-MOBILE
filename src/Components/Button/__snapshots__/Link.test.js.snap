// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#03989E",
        "paddingHorizontal": 5,
        "textAlign": "left",
      }
    }
  >
    testValue
  </Text>
</View>
`;
